gui/FunctionPlotterApp.cpp

```diff
@@ -12,11 +12,18 @@ FunctionPlotterApp::FunctionPlotterApp(
 	};
 
 	// widget for input
+	this->functionPlotter = new FunctionPlotter{};
 	FunctionPlotterInput *input = new FunctionPlotterInput{
-		window, renderer, 320, 450, 600, 500, "", 24, graph
+		window, renderer, 320, 450, 600, 500, "", 24, graph, this->functionPlotter
 	};
 
 	widgets.push_back(graph);
```

Comment:
I think `this->widgets.push_back(graph);` would be less confusing, I just wondered where it came from!
Something for cleanup #33 maybe?